### PR TITLE
#10 front

### PR DESCRIPTION
**Pull request by Ozonek95**
Please review the code.

### CODE REVIEW

File: src/main/java/pl/wilki/battleship/shipplacement/user/PutShipController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.messaging.handler.annotation.SendTo;
+import org.springframework.stereotype.Controller;
+
+/**
+ * @author Kacper Staszek
+ */
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/user/PutShipController.java:12]: missing java doc!

File: src/main/java/pl/wilki/battleship/shipplacement/user/PutShipController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  @SendTo("/topic/putMyShipMan")
+  public PutShipResult shot(PutShipObject putShipObject) throws Exception {
+    /*
+     * ONLY FOR TEST IMPLEMENTATION. HAVE TO CREATE JAVA CLASS THAT TAKES BOOLEAN VERTICAL/HORIZONTAL, AND ID, AND RETURNS PutShipResult valid Object!!!!
+     * in final version we would use class that returns list of fields to mark. PEACE
+     * */
+    int id = putShipObject.getId();
+    boolean shiftPressed = putShipObject.isShiftPressed();
+    List<Integer> toReturn = new ArrayList<>();
+    if (shiftPressed) {
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/user/PutShipController.java]: this method is way to big, you should consider divide it into few smaller ones

File: src/main/java/pl/wilki/battleship/shipplacement/user/PutShipObject.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package pl.wilki.battleship.shipplacement.user;
+
+public class PutShipObject {
+
+  private boolean shiftPressed;
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/user/PutShipObject.java:5]: missing javadoc!

File: src/main/java/pl/wilki/battleship/shipplacement/user/PutShipObject.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  public PutShipObject(int id, boolean shiftPressed) {
+    this.shiftPressed = shiftPressed;
+    this.id = id;
+  }
+
+  int getId() {
+    return id;
+  }
+
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/user/PutShipObject.java:19]: I don't think you need setters, not sure though

File: src/main/java/pl/wilki/battleship/shipplacement/user/PutShipResult.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package pl.wilki.battleship.shipplacement.user;
+
+import java.util.List;
+
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/user/PutShipResult.java:4]: crtl+alt+l

File: src/main/java/pl/wilki/battleship/web_socket_configuration/WebSocketConfig.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package pl.wilki.battleship.web_socket_configuration;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.simp.config.MessageBrokerRegistry;
+import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
+import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
+import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;
+
+@Configuration
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/web_socket_configuration/WebSocketConfig.java:9]: missing java doc!

File: src/main/resources/static/main.css [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    padding: 15px 20px;
+    text-align: center;
+    text-decoration: none;
+    display: inline-block;
+    font-size: 15px;
+    -webkit-transition-duration: 0.4s; /* Safari */
+    transition-duration: 0.4s;
+
+}
+
```
Comment (TwojaTata) [inline src/main/resources/static/main.css:41]: is this comment needed?

File: src/test/java/pl/wilki/battleship/board/BoardTest.java [hunk 8]
```diff
@@ -11,6 +11,69 @@
  */
 @Test
 public class BoardTest {
+
+  @DataProvider
+  public static Object[][] newDefaultBoardFieldsCheck() {
```
Comment (TwojaTata) [inline src/test/java/pl/wilki/battleship/board/BoardTest.java]: I have no idea what does this DataProvider, you should name it like "fieldsTypeToCheckDataProvider"

File: src/test/java/pl/wilki/battleship/board/BoardTest.java [hunk 9]
```diff
@@ -11,6 +11,69 @@
...
+        {50, FieldState.WATER},
+        {51, FieldState.WATER},
+        {49, FieldState.WATER},
+        {10, FieldState.WATER},
+        {90, FieldState.WATER},
+    };
+  }
+
+  @DataProvider
+  public static Object[][] setFieldCheck() {
```
Comment (TwojaTata) [inline src/test/java/pl/wilki/battleship/board/BoardTest.java]: same here, rename this method
Also, you could generate this data, not hard-code it. Simple data generating algorithm for data and you can check all possible combinations.

File: src/main/java/pl/wilki/battleship/Application.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
+package pl.wilki.battleship;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/Application.java:5]: java doc!

File: src/main/java/pl/wilki/battleship/shipplacement/random/RandomShipsController.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
+package pl.wilki.battleship.shipplacement.random;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.messaging.handler.annotation.SendTo;
+import org.springframework.stereotype.Controller;
+
+/**
+ * @author Ola Podorska
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/random/RandomShipsController.java:9]: java doc!

File: src/main/java/pl/wilki/battleship/shipplacement/random/ShipGenerator.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Ola Podorska
+ */
+
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/random/ShipGenerator.java:13]: unecessary class

File: src/main/java/pl/wilki/battleship/shipplacement/user/PutShipResult.java [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * @author Kacper Staszek
+ */
+class PutShipResult {
+
+  private List<Integer> fieldsOfShip;
+  private boolean valid;
+
+  public PutShipResult() {
```
Comment (TwojaTata) [inline src/main/java/pl/wilki/battleship/shipplacement/user/PutShipResult.java:14]: can't be more public than this

**Review by TwojaTata — CHANGES_REQUESTED** [review]
Dude, whole application is in this commit and you neme it "ShipGenerator tested"??!@?!!?@?!@?#!?@#!@#

File: src/main/java/pl/wilki/battleship/web_socket_configuration/WebSocketConfig.java (hunk 6 above)
Comment (atrzewik) [reply src/main/java/pl/wilki/battleship/web_socket_configuration/WebSocketConfig.java:9]: dalej brak

**Review by TwojaTata — APPROVED** [review]
gites, majonezus

**Review by atrzewik — CHANGES_REQUESTED** [review]

**Review by atrzewik — APPROVED** [review]

**Pull request merged**